.github/workflows/build-and-deploy.yml

```diff
@@ -0,0 +1,54 @@
+name: Build and Deploy
+
+on:
+  push:
+    branches:
+      - trigger-build
+      - main
+  workflow_dispatch: # Allows manual trigger
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v4
+
+      - name: Set up SSH for Git (deploy key)
+        run: |
+          mkdir -p ~/.ssh
+          echo "${{ secrets.SSH_PRIVATE_KEY }}" > ~/.ssh/id_ed25519
+          chmod 600 ~/.ssh/id_ed25519
+          ssh-keyscan github.com >> ~/.ssh/known_hosts
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Dependencies
+        run: |
+          python3 -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Fetch GitHub data
+        run: |
+          python3 gh-data.py
+
+      - name: Merge data sources
+        run: |
+          python3 merge-data.py
+
+      - name: Commit and Push Changes to `gh-pages`
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git checkout -b gh-pages
+          git add -f gh-data.json
+          git add -f gh-data-summary.json
+          git add -f merged-data.json
+          git add .
+          git commit -m "Update GitHub Pages with latest data"
+          git remote set-url origin git@github.com:${{ github.repository }}.git
+          git push origin gh-pages --force
```

Comment:
A force push seems unnecessary here.  Better to let it fail, I'd say.

Alternatively, you could use the GitHub pages action, like I do [here](https://github.com/martinthomson/asciitable/blob/main/.github/workflows/index.yml).  The advantage of that is that you don't end up tracking changes constantly.

Reply:
Now only `merged-data.json` is added.